src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -610,55 +936,25 @@ public SocketError Accept(byte[] socketAddress, ref int socketAddressLen, int ti
             Debug.Assert(timeout == -1 || timeout > 0, $"Unexpected timeout: {timeout}");
 
             SocketError errorCode;
-            if (SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
+            int observedSequenceNumber;
+            if (_receiveQueue.IsReady(this, out observedSequenceNumber) &&
+                SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
```

Comment:
Not sure I follow.  How would PerformSyncOperation behave if we hadn't observed the seq number previously?  It doesn't know whether it needs to retry or not.  It could always retry, but that's expensive as it involves a kernel call.